app/addons/databases/components.react.jsx

```diff
@@ -0,0 +1,271 @@
+          <td>{row.status.updateSeq()}</td>
+          <td>
+            <a className="db-actions btn fonticon-replicate set-replication-start" title={"Replicate "+name} href={"#/replication/"+encoded}></a>&#160;
+            <a className="db-actions btn icon-lock set-permissions" title={"Set permissions for "+name} href={"#/database/"+encoded+"/permissions"}></a>
+          </td>
+        </tr>
+      );
+    }
+  });
+
+  var GraveyardInfo = React.createClass({
+
+    componentDidMount : function () {
+      $(this.refs.myself.getDOMNode()).tooltip();
+    },
+
+    render : function () {
+      var row = this.props.row;
+      return (
+        <i className="js-db-graveyard icon icon-exclamation-sign" ref="myself" title={"This database has just " + row.status.numDocs() + " docs and " + row.status.numDeletedDocs() + " deleted docs"}></i>
+      );
+    }
+  });
+
+  var RightDatabasesHeader = React.createClass({
+
+    render : function () {
+      return (
+        <div className="header-right">
+          <AddDatabaseWidget />
+          <JumpToDatabaseWidget />
+        </div>
+      );
+    }
+  });
+
+  var AddDatabaseWidget = React.createClass({
+
+    onTrayToggle : function () {
+      var that = this;
+      this.refs.newDbTray.toggle(function (shown) {
+        if (shown) {
+          that.refs.newDbName.getDOMNode().focus();
+        }
+      });
+    },
+
+    onKeyUpInInput : function (e) {
+      if (e.which === 13) {
+        this.onAddDatabase();
+      }
+    },
+
+    onAddDatabase : function () {
+      var databaseName = $(this.refs.newDbName.getDOMNode()).val();
+      Actions.createNewDatabase(databaseName, this.refs.newDbTray.hide, FauxtonAPI.addNotification);
+    },
+
+    render : function () {
+
+      return (
+        <div className="button" id="add-db-button">
+          <a id="add-new-database" className="add-new-database-btn" href="#" onClick={this.onTrayToggle}><i className="header-icon fonticon-new-database"></i> Add New Database</a>
+          <ComponentsReact.Tray trayid="new-database-tray" ref="newDbTray" className="new-database-tray">
+            <span className="add-on">Add New Database</span>
+            <input id="js-new-database-name" type="text" onKeyUp={this.onKeyUpInInput} ref="newDbName" className="input-xxlarge" placeholder="Name of database" />
+            <a className="btn" id="js-create-database" onClick={this.onAddDatabase}>Create</a>
+          </ComponentsReact.Tray>
+        </div>
+      );
+    }
+  });
+
+  var JumpToDatabaseWidget = React.createClass({
+
+    componentDidMount : function () {
+      var that = this;
+      $(this.refs.searchDbName.getDOMNode()).typeahead({
+        source : databasesStore.getDatabaseNames(),
+        updater: function (item) {
+          that.jumpToDb(item);
+        }
+      });
+      // turn of browser autocomplete
+      $(this.refs.searchDbName.getDOMNode()).attr("autocomplete", "off");
```

Comment:
you can't - React will drop the attribute. And we're doing JQuery here anyway. As long as we don't tynamically update the input field I think we're very OK and won't break things. We just have to remember not to bind anything to the input dynamically via react. Otherwise there would be "data-autocomplete='off'" - but still: we're using JQuery anyway and data-... does at least have a similar chance of getting in the way...